flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/aggregate/AggregateUtil.scala

```diff
@@ -916,6 +916,10 @@ object AggregateUtil {
                   new DecimalMinAggFunction
                 case BOOLEAN =>
                   new BooleanMinAggFunction
+                case VARCHAR =>
+                  new StringMinAggFunction
+                case CHAR =>
```

Comment:
for 1 and 2, i will add max/minWithRetractAggFunction, ut, and integration test.
for 3, we do support CHAR type in SQL, not in table api.